rskj-core/src/main/java/co/rsk/rpc/modules/eth/subscribe/AccountOverride.java

```
@@ -0,0 +1,133 @@
+package co.rsk.rpc.modules.eth.subscribe;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import org.ethereum.core.Repository;
+import org.ethereum.vm.DataWord;
+
+import java.math.BigInteger;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+public class AccountOverride {
+    private BigInteger balance;
+    private Long nonce;
+    private byte[] code;
+    private Map<DataWord, DataWord> state;
+    private Map<DataWord, DataWord> stateDiff;
+    private RskAddress address;
+    //TODO movePrecompile to address
+    private RskAddress movePrecompileToAddress;
+
+    public BigInteger getBalance() {
+        return balance;
+    }
+
+    public void setBalance(BigInteger balance) {
+        this.balance = balance;
+    }
+
+    public Long getNonce() {
+        return nonce;
+    }
+
+    public void setNonce(Long nonce) {
+        this.nonce = nonce;
+    }
+
+    public byte[] getCode() {
+        return code;
+    }
+
+    public void setCode(byte[] code) {
+        this.code = code;
+    }
+
+    public Map<DataWord, DataWord> getState() {
+        return state;
+    }
+
+    public void setState(Map<DataWord, DataWord> state) {
+        this.state = state;
+    }
+
+    public Map<DataWord, DataWord> getStateDiff() {
+        return stateDiff;
+    }
+
+    public void setStateDiff(Map<DataWord, DataWord> stateDiff) {
+        this.stateDiff = stateDiff;
+    }
+
+    public RskAddress getAddress() {
+        return address;
+    }
+
+    public void setAddress(RskAddress address) {
+        this.address = address;
```

question:

Can the address have any value? Can it be null?

---

Nope, there is an if condition to make it fail during the apply but if you prefer I can add it to the constructor and make the validation there.

---

I would say that these validations should go to the model, no?
Do you think we should put it elsewhere and let it fail in a later state?